data/virt_autotest/fetch_logs_from_guest.sh

```diff
@@ -0,0 +1,302 @@
+        mkdir -p ${logs_folder}/${guest_transformed}
+        echo -e "guestfish -r -d ${guest_domain} -m ${guest_device} copy-out ${logs_fetched} ${logs_folder}/${guest_transformed}"
+        guestfish -r -d ${guest_domain} -m ${guest_device} copy-out ${logs_fetched} ${logs_folder}/${guest_transformed}
+        if [[ $? -eq 0 ]];then
+           echo -e "Copied out ${logs_fetched} from ${guest_domain} successfully."
+           return 0
+        else
+           echo -e "Try again to mount ${guest_device}:/:subvol=@ with guestfish."
+           echo -e "guestfish -r -d ${guest_domain} -m ${guest_device}:/:subvol=@ copy-out ${logs_fetched} ${logs_folder}/${guest_transformed}"
+           guestfish -r -d ${guest_domain} -m ${guest_device}:/:subvol=@ copy-out ${logs_fetched} ${logs_folder}/${guest_transformed}
+           if [[ $? -eq 0 ]];then
+              echo -e "Copied out ${logs_fetched} from ${guest_domain} successfully."
+              return 0
+           else
+              echo -e "Failed to copy out ${logs_fetched} from ${guest_domain}."
+              return 1
+           fi
+        fi
+}
+
+#Power off virtual machine if necessary and remove logs folder in it by using libguestfs
+function remove_logs_folder_from_guest_via_libguestfs() {
+        local guest_domain=$1
+        local logs_folder=$2
+        local ret_result=0
+
+        echo -e "Going to remove ${logs_folder} from ${guest_domain} via libguestfs"
+        local guest_device=`find_disk_hosts_filesystem ${guest_domain} ${logs_folder}`
+        echo -e "guestfish -w -d ${guest_domain} -m ${guest_device} rm-rf ${logs_folder}"
+        guestfish -w -d ${guest_domain} -m ${guest_device} rm-rf ${logs_folder}
+        ret_result=$?
+        if [[ ${ret_result} -ne 0 ]];then
+	   echo -e "Power off ${guest_domain} to make read-write access possible via libguestfs."
+           virsh destroy ${guest_domain}
+           echo -e "guestfish -w -d ${guest_domain} -m ${guest_device} rm-rf ${logs_folder}"
+           guestfish -w -d ${guest_domain} -m ${guest_device} rm-rf ${logs_folder}
+           ret_result=$?
+           if [[ ${ret_result} -ne 0 ]];then
+              echo -e "Try again to mount ${guest_device}:/:subvol=@ with guestfish."
+              echo -e "guestfish -w -d ${guest_domain} -m ${guest_device}:/:subvol=@ rm-rf ${logs_folder}"
+              guestfish -w -d ${guest_domain} -m ${guest_device}:/:subvol=@ rm-rf ${logs_folder}
+              ret_result=$?
+           fi
+        fi
+
+        virsh start ${guest_domain}
+        return ${ret_result}
+}
+
+#Fetch logs from virtual machine to local host via ssh firstly. Will resort to libguestfs tools if ssh connection is broken
+function fetch_logs_from_guest() {
+	local guest_domain=$1
+	local guest_ipaddr=$2
+	local logs_folder=$3
+	shift
+	shift
+	shift
+	local extra_logs=($@)
+
+	local ret1=0
+	fetch_logs_from_guest_via_ssh ${guest_domain} ${guest_ipaddr} ${logs_folder}
```

Comment:
Why only remove when fetch_logs_from_guest_via_libguestfs is successful, but no removal when fetch_logs_from_guest_via_ssh is successful?

Reply:
It is done by just one line:
${sshpass_ssh_cmd} rm -f -r ${logs_folder}